python/pyspark/tests.py

```diff
@@ -874,6 +874,15 @@ def test_sortByKey_uses_all_partitions_not_only_first_and_last(self):
             for size in sizes:
                 self.assertGreater(size, 0)
 
+    def test_pipe_functions(self):
+        data = ['1', '2', '3']
+        rdd = self.sc.parallelize(data)
+        with QuietTest(self.sc):
+            self.assertRaises(Py4JJavaError, rdd.pipe('cc').collect)
+        result = rdd.pipe('cat').collect()
```

Comment:
Could we have an option for this? At least, we should have an option to support `grep` on a partition without any matching lines.